sqla_yaml_fixtures.py

```
@@ -123,9 +129,16 @@ def load(ModelBase, session, fixture_text):
     # make sure backref attributes are created
     sqlalchemy.orm.configure_mappers()
 
+    # Data should be sequence of entry per mapper name
+    # to enforce that FKs (__key__ entries) are defined first
     data = yaml.load(fixture_text)
+    if not isinstance(data, list):
+        raise ValueError('top level YAML should be sequence (list)')
+
     store = Store()
-    for model_name, instances in data.items():
+    for model_entry in data:
+        # Model entry is dict with one item
+        model_name, instances = model_entry.popitem()
```

Better check and raise an error if the dict contains more than one entry.

---

ok, you're thorough 👍

---

Propose to also skip zero length model entries. I have a prog that generates a `sqla_fixture` from SQLAlchemy model. If the table is empty it generates just an empty `- Name:`. Ok?

---

I guess empty is ok. it is not something like will surprise an user with a silent failure.

---

Looking ahead: a generic `sql_yaml_fixtures.dump()` would be very useful for e.g. a generic `import/export` facility in e.g. a Flask app. But this is a separate feature maybe for later...

---

Also added test for empty case...